.pre-commit-config.yaml

```diff
@@ -83,7 +83,7 @@ repos:
     rev: v3.15.2
     hooks:
       - id: pyupgrade
-        args: [--py38-plus, --keep-runtime-typing]
+        args: [--py310-plus, --keep-runtime-typing]
```

Comment:
`pyupgrade` also affects syntax other than typing. Therefore, I was concerned about possible syntax errors in python<3.10. However, I decide it is not too late to take care of it after something happen.

Reply:
We could drop `pyupgrade` altogether and use ruff instead, see with https://docs.astral.sh/ruff/rules/#pyupgrade-up. This way it may be easier to control different options rather than having it as a kind of all-or-nothing with `pyupgrade`. This would be similar to dropping isort from #5712.

Reply:
This has been reverted to `--py38-plus`. With the `--keep-runtime-typing` option, `ruff` will treat fixes such as [UP006](https://docs.astral.sh/ruff/rules/non-pep585-annotation/) as "unsafe" and won't automatically apply the fixes. This is a good thing, since in some (very few/relatively uncommon) cases we need the runtime behavior (e.g. for some aliases we need to use `Tuple`, and not `tuple`). With the `--py38-plus` oftion ruff will simply complain about using `Tuple` but won't autofix this.